src/server/json_family.cc

```diff
@@ -692,8 +702,135 @@ OpResult<vector<OptSizeT>> OpArrTrim(const OpArgs& op_args, string_view key, str
   return vec;
 }
 
+OpResult<vector<OptLong>> OpArrIndex(const OpArgs& op_args, string_view key,
+                                     JsonExpression expression, const json& search_val,
+                                     int start_index, int end_index) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  vector<OptLong> vec;
+  auto cb = [&](const string_view& path, const json& val) {
+    size_t search_start_index;
+    size_t search_end_index;
+    if (!val.is_array()) {
+      vec.emplace_back(nullopt);
+      return;
+    }
+
+    if (val.empty()) {
+      vec.emplace_back(-1);
+      return;
+    }
+
+    if (start_index < 0) {
+      int idx = val.size() + start_index;
+      if (idx < 0) {
```

Comment:
what's the point of doing `idx %= val.size()` for negative indices?  it's not a common op, so some comments would be useful.